### PR TITLE
fix: vDisksPDisks is required in expert mode

### PR DESCRIPTION
**Pull request by Raubzeug**
### Test Status: <span style="color: orange;">⚠️ FLAKY</span>
  📊 [Full Report](https://ydb-platform.github.io/ydb-embedded-ui/4148/?t=1784730727740)

  | Total | Passed | Failed | Flaky | Skipped |
  |:-----:|:------:|:------:|:-----:|:-------:|
  | 768 | 765 | 0 | 3 | 0 |

  😟 No changes in tests. 😕

  ### Bundle Size: ✅
  Current: 64.43 MB | Main: 64.43 MB
  Diff: +0.84 KB (0.00%)

  ✅ Bundle size unchanged.

  <details>
  <summary>ℹ️ CI Information</summary>

  - Test recordings for failed tests are available in the full report.
  - Bundle size is measured for the entire 'dist' directory.
  - 📊 indicates links to detailed reports.
  - 🔺 indicates increase, 🔽 decrease, and ✅ no change in bundle size.
  </details>

<h3>Greptile Summary</h3>

This PR fixes a bug where the `VDisksPDisks` column was not being shown in expert mode when the user had previously deselected it in their saved column preferences.

- **`useSelectedColumns.ts`**: Introduces a `stickyColumnsIds` parameter (defaults to `requiredColumnsIds` for backward compatibility), splitting \"pinned to front\" (sticky) from \"always shown\" (required). Previously all required columns were both sticky and required; now required-but-not-sticky columns preserve their user-saved position while still being forced visible.
- **`hooks.ts`**: Separates `stickyColumns` (always-pinned columns) from `requiredColumns` (sticky columns + `VDisksPDisks` in expert mode), so that `VDisksPDisks` becomes required (always selected) in expert mode without being forcibly repositioned to the front.
- Tests are updated and a new unit test in `useSelectedColumns.test.ts` directly validates that a required-but-not-sticky column stays in its saved position.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the change is narrowly scoped to column-selection logic, all existing callers of `useSelectedColumns` get unchanged behavior via the defaulted `stickyColumnsIds` parameter, and the bug fix is directly validated by both updated and new tests.

The required/sticky split is well-reasoned: existing callers don't pass `stickyColumnsIds` so they fall back to `requiredColumnsIds` and keep identical behavior. The Storage Groups hook is the only place that passes distinct values, and the four affected test cases cover normal mode, expert mode with defaults, expert mode with a previously deselected column, and the edge case where both disk columns were deselected. No regressions were introduced.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/utils/hooks/useSelectedColumns.ts | Adds `stickyColumnsIds` optional parameter (defaults to `requiredColumnsIds`) to decouple "always selected" from "pinned to front"; backward-compatible for all existing callers. |
| src/containers/Storage/PaginatedStorageGroupsTable/columns/hooks.ts | Refactors required-column logic into separate `stickyColumns` and `requiredColumns` memos; VDisksPDisks is now required (not sticky) in expert mode, fixing the bug where user-deselected VDisksPDisks disappeared in expert mode. |
| src/containers/Storage/PaginatedStorageGroupsTable/columns/__test__/hooks.test.ts | Tests updated/renamed to reflect the new required-vs-sticky semantics; covers non-expert, expert with defaults, expert with saved deselection, and the edge case where both disk columns were disabled. |
| src/utils/hooks/__test__/useSelectedColumns.test.ts | New test validates that a required-but-not-sticky column preserves its user-saved position while being forced selected; existing tests continue to pass with the default parameter behaviour. |

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[useStorageGroupsSelectedColumns] --> B[stickyColumns\nGroupId, PoolName, Erasure, Used\n+ Degraded/DiskSpace by filter]
    A --> C{isStorageExpertMode\n&& isVDisksPDisksColumnAvailable?}
    C -- Yes --> D[requiredColumns =\nstickyColumns + VDisksPDisks]
    C -- No --> E[requiredColumns = stickyColumns]
    B --> F[useSelectedColumns\ncolumns, storageKey, titles,\ndefaultColumns, requiredColumns, stickyColumns]
    D --> F
    E --> F
    F --> G{isSticky?\nrequired && in stickyColumnsIds}
    G -- Yes --> H[sticky: 'start'\npinned to front]
    G -- No --> I[sticky: undefined\npreserves saved position\nbut forced selected=true]
    F --> J{shouldUseExpertDisksColumn?}
    J -- Yes --> K[Filter VDisks out of columnsToShow\nHide VDisks from columnsToSelect]
    J -- No --> L[Return columns as-is]
```

<sub>Reviews (2): Last reviewed commit: ["fix: VDisksPDisks is required in expert ..."](https://github.com/ydb-platform/ydb-embedded-ui/commit/c2634517c82f26ce1fc66a43b01024ff47f63214) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=46314145)</sub>

### CODE REVIEW

**Comment by Raubzeug** [comment]
@greptile review

File: src/containers/Storage/PaginatedStorageGroupsTable/columns/hooks.ts [hunk 1]
```diff
@@ -69,20 +69,26 @@ export function useStorageGroupsSelectedColumns({
...
+            sticky.push(STORAGE_GROUPS_COLUMNS_IDS.DiskSpace);
         }
 
-        return required;
+        return sticky;
     }, [visibleEntities]);
 
+    const requiredColumns = React.useMemo(() => {
+        return isStorageExpertMode && isVDisksPDisksColumnAvailable
+            ? [...stickyColumns, STORAGE_GROUPS_COLUMNS_IDS.VDisksPDisks]
```
[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/containers/Storage/PaginatedStorageGroupsTable/columns/hooks.ts:88]: Thats ok!

**Review by astandrik — APPROVED** [review]
